cachi2/core/package_managers/yarn/project.py

```diff
@@ -73,8 +81,23 @@ def registry_server_for_scope(self, scope: str) -> str:
 
         return registry or self.registry_server
 
+    @staticmethod
+    def load_defaults(source_dir: RootedPath) -> dict[str, Any]:
+        """Load Yarn RC defaults.
+
+        :param source_dir: directory context used to determine the right yarn binary to execute,
+                           e.g. yarn v1 doesn't support the 'config' subcommand
+        :returns: dictionary containing Yarn RC defaults if form of {"<option>": "<default_value>"}
+        """
+
+        yarn_config = json.loads(run_yarn_cmd(["config", "--json"], source_dir))
```

Comment:
You may need to pass `YARN_IGNORE_PATH=true` as an env var here. Otherwise the globally installed `yarn` will replace itself with whatever the user has configured as `yarnPath` in the config file. And that can easily be something malicious.

(I'm assuming `load_defaults()` happens before we've had a chance to set `ignorePath` in the config file)

Reply:
Oh, I was blindly relying on the directory context. I did wonder whether that's a reliable way of doing it. Thanks, I will change it accordingly.

Reply:
The directory context probably would be enough if the .yarnrc.yml file already had `ignorePath: true` set. But if loading the defaults is part of YarnRC instantiation, and we use YarnRC to overwrite the .yarnrc.yml, then that's a bit of a chicken and egg proglem (thankfully it's not, thanks to the env vars)

Reply:
@chmeliik question: In my local setup I have (mistakenly) installed Yarn v1 globally and v4 in the user repo. Do we check early and refuse Yarn <v2+? Otherwise I'd have to add more handling and checks to make sure right Yarn is executed because v1 doesn't have the `config` subcommand.

Reply:
I think we just assume that the global yarn is the [corepack](https://nodejs.org/api/corepack.html) yarn shim. Which is true in the cachi2 container, but not necessarily elsewhere